CMake/utils/kwiver-utils-msvc.cmake

```diff
@@ -0,0 +1,109 @@
+# used in batch files to signal the build type folder to use in batch scripts
+#-
+
+
+if(POLICY CMP0057)
+  cmake_policy(SET CMP0057 NEW)
+endif()
+
+function(kwiver_setup_msvc_env)
+  
+  foreach(setup_batch ${ARGN})
+    list(APPEND SETUP_BATCH_FILES "${setup_batch}")
+  endforeach()
+  
+  foreach(setup_batch ${SETUP_BATCH_FILES})
+    #message(STATUS "Extracting environment from ${setup_batch}")
+    get_filename_component(batch_dir ${setup_batch} DIRECTORY)
+    file(READ "${setup_batch}" contents)
+    # Convert file contents into a CMake list
+    # Where each element in the list is one line of the file
+    string(REGEX REPLACE ";" "\\\\;" contents "${contents}")
+    string(REGEX REPLACE "\n" ";" contents "${contents}")
+    foreach(line ${contents})
+      string(STRIP "${line}" line)
+      string(TOLOWER "${line}" search)
+      # Ignore comments
+      string(FIND "${search}" "::" idx)
+      if("${idx}" GREATER -1)
+        continue()
+      endif()
+      string(FIND "${search}" "rem" idx)
+      if("${idx}" GREATER -1)
+        continue()
+      endif()
+      string(FIND "${search}" "set config" idx)
+      if("${idx}" GREATER -1)
+        continue()
+      endif()
+      # check for set
+      string(FIND "${search}" "set " idx)
+      string(FIND "${search}" "=" edx)
+      if(NOT "${idx}" EQUAL 0 OR ${edx} EQUAL -1)
+        continue()
+      endif()
+      
+      #message(STATUS "I cleaning ${line}")
+      string(LENGTH "${line}" length)
+      math(EXPR _vardx "${edx}-4")
+      math(EXPR _valdx "${edx}+1")
+      string(SUBSTRING "${line}" 4 ${_vardx} _var) # what are we setting?
+      string(SUBSTRING "${line}" ${_valdx} ${length} _val) # what are we setting it to?
+      # replace batch terms with msvc terms
+      string(REPLACE "%config%" "$(Configuration)" _val "${_val}")
+      string(REPLACE "%~dp0" "${batch_dir}" _val "${_val}")
+      # remove any recursive sets
+      string(REPLACE "%${_var}%" "" _val "${_val}")
+      #message(STATUS "I am setting ${_var} to ${_val}")
+      # Keep track of setting the same variable over and over
+      if(NOT ${_var} IN_LIST _env_variables)
+        list(APPEND _env_variables "${_var}")
+      endif()
+      list(APPEND "_env_${_var}" "${_val}")
+    endforeach()
+  endforeach()
+  # Consolidate setting env variables into one line    
+  foreach(_env_var ${_env_variables})
+    #message(STATUS "processing ${_env_var}")
+    set(MSVC_ENV "${MSVC_ENV}${_env_var}=")
+    foreach(_env_val ${_env_${_env_var}})
+      set(MSVC_ENV "${MSVC_ENV}${_env_val};")
+    endforeach()
+    set(MSVC_ENV "${MSVC_ENV}\n")
+  endforeach()
+  #message(STATUS "Setting MSVC environment to ${MSVC_ENV}")
+    
+  get_property(executables GLOBAL PROPERTY kwiver_executables)
+  get_property(executables_paths GLOBAL PROPERTY kwiver_executables_paths)
+  #message(STATUS "Create msvc files for these executables : ${executables}")
+  list(LENGTH executables stop)
+  math(EXPR stop "${stop}-1")
```

Comment:
It might be worth some comments about what's going on in here. I don't fully follow it.

Reply:
ok